datasources/src/main/java/org/opensearch/sql/datasources/glue/GlueDatasourceFactory.java

```diff
@@ -0,0 +1,66 @@
+package org.opensearch.sql.datasources.glue;
+
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.Map;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.datasource.model.DataSource;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.datasources.utils.DatasourceValidationUtils;
+import org.opensearch.sql.storage.DataSourceFactory;
+
+@RequiredArgsConstructor
+public class GlueDatasourceFactory implements DataSourceFactory {
+
+  private final Settings pluginSettings;
+
+  // Glue configuration properties
+  public static final String GLUE_AUTH_TYPE = "glue.auth.type";
+  public static final String GLUE_ACCOUNT_ID = "glue.auth.account_id";
+  public static final String GLUE_ROLE_ARN = "glue.auth.role_arn";
+  public static final String GLUE_REGION = "glue.auth.region";
+
+  public static final String FLINT_URI = "emr.datasource.flint.uri";
+  public static final String FLINT_AUTH = "emr.datasource.flint.auth";
```

Comment:
could you add the meaning of all these configs to doc?

Reply:
will add a new page for glue datasource connector page.